xls/modules/zstd/buffer.x

```diff
@@ -0,0 +1,363 @@
+// Copyright 2023 The XLS Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This file contains implementation of a simple buffer that can store
+// data from multiple transfers over the XLS channel and glue them together.
```

Comment:
since it is meant to be used with channel, it would be nice to provide a `test _proc` that demonstrate (and tests!) idiomatic usage scenarios.

Reply:
I provided a simple use-case scenario in a separate commit